### ISSUE TITLE
Custom ORACLE_HOME for each database

### ISSUE THREAD

**Issue by Rendanic** [oravirt/ansible-oracle#18]
I like to add support for custom ORACLE_HOME for each Datanase. I mentioned it in a pull request and you wrote that this is really welcome.

This is currently possible, but not with custom path for ORACLE_HOME. 
There are a lot of tasks and templates with `'{{ oracle_base }}/{{ item.oracle_version_db }}/{{ item.home }}'` except `'{{ oracle_home_db }}'`. I think you added this variable to the defaults at a later time and didn't fixed the tasks.

I like to add the variable oracle_home to oracle_databases and change the following line in oradb-create/defaults/main.yml:
`oracle_home_db: "{% if item.oracle_home is defined %}{{ item.oracle_home }}{% else %}{{ oracle_base }}/{{ item.oracle_version_db }}/{{ item.home }}{% endif %}"`

This is compatible against all existing configurations and give us tha ability to define a custom ORACLE_HOME for each database. As a  side effect. You also get the ability to define a custom ORACLE_HOME for all databases, which is not hard coded anymore when oracle_home_db is defined as a variable. 

I'll start the work on this change, if this is fine for you.

**Comment by oravirt** [comment]
This is fine.
I'm going to fix the other hardcoded paths at a later stage.

**Comment by Rendanic** [comment]
It took me some time to find a working solution...
One major problem is using `with_items` and `with_together` in the roles.
The following solution looks not very nice but it is working. I added linefeeds here for better readability, because the whole definition is written in 1 line...

```
oracle_home_db: 
{% if item.0 is defined %}
  {% if item.0.oracle_home is defined %}
    {{ item.0.oracle_home}}
  {% else %}
    {{ oracle_base }}/{{ item.0.oracle_version_db }}/{{ item.0.home }}
  {% endif %}
{% else %}
  {% if item.oracle_home is defined %}
    {{ item.oracle_home }}
  {% else %}
    {{ oracle_base }}/{{ item.oracle_version_db }}/{{ item.home }}
  {% endif %}
{% endif %}
```

This should work for other variables in defaults as well.

**Comment by oravirt** [comment]
Fixed as of eda5384
